### DISCUSSION TITLE
`delete_variable()` and `detach_function()`

### DISCUSSION THREAD

**Discussion by gvisca44** [VasilKalchev/LiquidMenu#66] [Ideas]
Forgive me ... first post on Github.  Nice library ... 

Given we have the facility to add a variable - might it be nice to delete too ?

Also, given we can attach a function - ability to detach might be nice ?

Cheers, 

Glenn.

**Comment by VasilKalchev** [comment]
Hello Glenn.

What use case do you have in mind for this functionality?

I'm hesitant to implement these methods without a use case, since a `remove()` method will have to rearrange the array in which the added variables are stored.

**Comment by gvisca44** [comment]
Oh !  almost 12 months since I posted this.  

I am attempting to implement with a pretty deep menu structure - to display and update many values that ultimately get stored in EEPROM (measured in 10s or 100s of stored parameters).  

At the time I was creating the prototype code I think I was attempting to avoid code bloat by adding/removing or attaching/detaching different variables/functions based on the parameter being updated (rather than creating a new object for each one).

Glenn.

**Reply by VasilKalchev** [reply]
Thanks for the reply.

I haven't though of that. I agree, it would be useful to have these methods.
For a future release, I'm thinking of attempting to change the static arrays into some dynamic data type, which will solve this problem in a clean way.

I hope you found a solution around this problem.

Regards,
Vasil
